tests/configm_test.sh

```diff
@@ -87,17 +96,26 @@ function test_save_config_file_check_save_failures()
   assert_equals_helper "Should return ENOENT, because '.config' != '.configuration'" "$LINENO" "$?" "2"
   rm .configuration
 
-  cd "$current_path"
+  cd "$current_path" || {
+    fail 'Was not able to move back from temp directory'
+    return 2 #ENOENT
```

Comment:
@kwy95 Isn't there a situation where cd would fail with error code 1 instead of 2?

Reply:
Yes; in fact, it usually does. From the bash reference the builtin functions return exit code of 2 to "indicate incorrect usage", and I haven't been able to produce any error code that isn't 1 from `cd`. I used 2 here based on the output to `errno -l`, that defines 1 and 2 as:
```
EPERM 1 Operation not permitted
ENOENT 2 No such file or directory
```
I think that maybe 1 would be a better option as in bash it's used as a catch-all error, this would also make it consistent with my choice to use 1 as the default of `exit_msg`.
I am open to suggestions of a more appropriate error code if anyone has suggestions.

Further reading:
https://www.gnu.org/software/bash/manual/html_node/Exit-Status.html
https://tldp.org/LDP/abs/html/exitcodes.html

Reply:
wait, the builtin functions do not follow the errno convention? if you cd to a folder that does not exist you don't get the error 2?

>  I am open to suggestions of a more appropriate error code if anyone has suggestions.

I think the best way would be to just propagate the error code, without hardcoding it. Is there some sort of technical challenge for doing this? because that's what you did in the exit_msg function

Reply:
They do not, trying to cd to a folder that doesn't exist just gives you error code 1. The errno convention is apparently target more at C and C++ files, and bash functions don't usually conform. As I said previously for the built-ins error code 2 has a strictly different meaning from the one defined in errno.

I do propagate from outside of this function, I will set the default to `$?` which should give the same result.

Reply:
I overlooked the fact this refers to the tests section. In the case of tests we don't want to exit as discussed above.
Here we can't simply `return $?` as at that point it will be referencing the exit code of the `fail` command. I could add a variable, but it will add a lot of clutter.
Any suggestions are welcome!

Reply:
@kwy95 maybe we can have something similar to `exit_msg` in the `tests/utils.sh`, but dedicated to tests. Maybe `fail_and_cancel_test`